### PR TITLE
extract rar files

### PR DESCRIPTION
**Pull request by zaidalyafeai**
Unfortunately, I didn't find any native python libraries for extracting rar files. The user has to manually install `sudo apt-get install unrar`. Discussion with @yjernite is in the slack channel.

### CODE REVIEW

File: src/datasets/utils/file_utils.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
 from zipfile import ZipFile, is_zipfile
 
 import numpy as np
+import rarfile
```
Comment (lhoestq) [inline src/datasets/utils/file_utils.py:25]: This import would make the library impossible to import for people that don't have `rarfile` installed.

To make it an optional import you can do the same as what we did for pytorch at the top of the `file_utils.py` file.

The idea is to have a boolean `_rarfile_available` that is True when `rarfile` is installed. Then before any call to `rarfile` you first have to check if it is avaibale using `_rarfile_available`.

**Review by lhoestq — COMMENTED** [review]
Thanks !

I just left a comment about making rarfile an optional dependency

File: src/datasets/utils/file_utils.py (hunk 1 above)
Comment (zaidalyafeai) [reply src/datasets/utils/file_utils.py:25]: Thanks @lhoestq , I also added `rarfile` to `setup.py` file, does that make sense or I should remove it from there ?

Comment (zaidalyafeai) [reply src/datasets/utils/file_utils.py]: Refactored everything. I tested with and without `rarfile` and it seems to work, also tested on a dataset that I am adding with rar files. I also added an exception if the file is `rar` but the user doesn't have `rarfile` installed.

Comment (lhoestq) [reply src/datasets/utils/file_utils.py]: Nice thanks !

And yes we should have rarfile in the `tests` requirements, thanks for adding it

**Review by lhoestq — APPROVED** [review]
Awesome thank you !

**Pull request merged**